python/src/multi_agent_orchestrator/agents/chain_agent.py

```
@@ -62,7 +62,7 @@ async def process_request(
 
             except Exception as error:
                 Logger.error(f"Error processing request with agent {agent.name}:{str(error)}")
-                raise f"Error processing request with agent {agent.name}:{str(error)}"
+                raise f"Error processing request with agent {agent.name}:{str(error)}" from error
```

Exceptions should propagate the original error for complete stack trace.